enable-vcp-image/enable-vcp-scripts/daemonset_pod.sh

```diff
@@ -70,21 +98,27 @@ if [ $? -eq 0 ]; then
     if [ $? -eq 0 ]; then
         echo "[INFO] Existing vsphere.conf file is copied to" /host/tmp/$POD_NAME/vsphere.conf
     else
-        echo "[ERROR] Failed to back up vsphere.conf file at " /host/tmp/$POD_NAME/vsphere.conf
+        ERROR_MSG="Failed to back up vsphere.conf file at " /host/tmp/${POD_NAME}/vsphere.conf
+        echo "[ERROR] ${ERROR_MSG}"
+        update_VcpConigStatus "$POD_NAME" "$PHASE" "FAILED" "$ERROR_MSG"
         exit $ERROR_FAIL_TO_BACKUP_FILE
     fi
 fi
 
 # locate and back up manifest files and kubelet service configuration file.
 file=/host/$k8s_secret_kubernetes_api_server_manifest
 backupdir=/host/tmp/$POD_NAME
-locate_validate_and_backup_files $file $backupdir
+locate_validate_and_backup_files $file $backupdir $POD_NAME
 
 file=/host/$k8s_secret_kubernetes_controller_manager_manifest
-locate_validate_and_backup_files $file $backupdir
+locate_validate_and_backup_files $file $backupdir $POD_NAME
 
 file=/host/$k8s_secret_kubernetes_kubelet_service_configuration_file
-locate_validate_and_backup_files $file $backupdir
+locate_validate_and_backup_files $file $backupdir $POD_NAME
```

Comment:
update_vcpConfigStatus is called inside locate_validate_and_backup_files.
I can see three calls to this function here.
Does that mean 3 time status will be updated?

I see that update_vcpStatus is getting called in multiple functions.
I would rather put all update_vcpStatus calls in main flow here and now in the individual functions.